File: test/Microsoft.AspNet.Mvc.Core.Test/Logging/NullLogger.cs
// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.

using System;
using Microsoft.Framework.Logging;

namespace Microsoft.AspNet.Mvc
{
    public class NullLogger : ILogger
    {
        public static NullLogger Instance = new NullLogger();

        public IDisposable BeginScope(object state)
        {
            return NullDisposable.Instance;
        }

        public void Write(TraceType eventType, int eventId, object state, Exception exception, Func<object, Exception, string> formatter)
        {
        }

        public bool IsEnabled(TraceType eventType)
        {
            return false;
        }
    }
}

File: clients/java/signalr/src/main/java/com/microsoft/signalr/NullLogger.java
// Copyright (c) .NET Foundation. All rights reserved.
// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.

package com.microsoft.signalr;

class NullLogger implements Logger {
    @Override
    public void log(LogLevel logLevel, String message) { }

    @Override
    public void log(LogLevel logLevel, String formattedMessage, Object... args) { }
}
